### PR TITLE
Automatically generate TOC for utils readme

### PR DESCRIPTION
**Pull request by joellabes**
This is a:
- [ ] bug fix PR with no breaking changes — please ensure the base branch is `master`
- [x] new functionality — please ensure the base branch is the latest `dev/` branch
- [ ] a breaking change — please ensure the base branch is the latest `dev/` branch

## Description & motivation
I noticed today that there were a handful of macros missing from the readme, and that's _after_ I've also merged a few PRs to add other missing ones 😬 

This fixes it forever by adding a GitHub action to generate a TOC based on the markdown headers, embedded between the `<!--ts-->` and `<!--te-->` comment tags. 

Snazzy.

### CODE REVIEW

File: .github/workflows/create-table-of-contents.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# This is a basic workflow to help you get started with Actions
+
+name: Update table of contents
+
+# Controls when the workflow will run
+on:
+  push:
+    branches: [main]
```
Comment (joellabes) [inline .github/workflows/create-table-of-contents.yml:8]: Maybe this should be all branches?

**Comment by joellabes** [comment]
@dbeatty10 I forgot I never merged this! Could you have a peek?

**Comment by dbeatty10** [comment]
> @dbeatty10 I forgot I never merged this! Could you have a peek?

@joellabes There some merge conflicts that need to be resolved. Want to take a stab at resolving those and I can give this a review afterwards?

**Comment by joellabes** [comment]
> There some merge conflicts that need to be resolved

good to go other than the fact that redshift is still playing up!

**Review by dbeatty10 — APPROVED** [review]
Let's give it a go!

Since this is a meta change to the code base rather than an actual change, I think it's fine to merge even though there's permissions errors with Redshift right now.

**Pull request merged**